Doc/library/stdtypes.rst

```diff
@@ -2392,6 +2392,196 @@ that ``'\0'`` is the end of the string.
 
+
+.. index::
+   single: formatted string literal
+   single: interpolated string literal
+   single: string; formatted literal
+   single: string; interpolated literal
+   single: f-string
+   single: {} (curly brackets); in formatted string literal
+   single: ! (exclamation); in formatted string literal
+   single: : (colon); in formatted string literal
+   single: = (equal); in debug string literal
+.. _f-string-formated-string-literal:
+
+``f-string``-Formatted String Literal
+-------------------------------------
+.. versionadded:: 3.6
+
+A :dfn:`formatted string literal` or :dfn:`f-string` is a string literal
+that is prefixed with ``'f'`` or ``'F'``.  These strings may contain
+replacement fields, which are expressions delimited by curly braces ``{}``.
+While other string literals always have a constant value, formatted strings
+are really expressions evaluated at run time.
+
+This adds flexibility to how strings can be formatted::
+
+   >>> name = "Fred"
+   >>> f"He said his name is {name!r}."
+   "He said his name is 'Fred'."
+
+The above is an example and will become clear as you read below.
+
+Everything outside the curly braces is treated as literally with the exception
+of double braces which are treated as one brace::
+
+   >>> f"This is a string"
+   'This is a string.'
+   >>> f'{{'
+   '{'
+   >>> F"{{"
+   '}'
+   >>> F'{{ }}'
+   '{ }'
+
+A single opening curly bracket, ``'{'``, marks a replacement field that starts with
+a python expression.::
+
+   >>> f"4 * 3 is {4 * 3}"
+   '4 * 3 is 12'
+
+The above is quite contrived and a more useful use in code could be::
+
+   >>> a = 4
+   >>> b = 3
+   >>> f"The product of {a} and {b} is {a * b}"
+   'The product of 4 and 3 is 12'
+
+Or more simply::
+
+   >>> a = 4
+   >>> b = 3
+   >>> f"The product is {a * b}"
+   'The product is 12'
+
+Methods can also be used::
+
+   >>> string = "hello, world!"
+   >>> f"{string.title()}"
+   'Hello, World!'
+   >>> def make_title(input):
+   ...     return input.title()
+   ...
+   >>> f"{make_title(string)}"
+   'Hello, World!'
+
+Any variable that can be presented as a string can be used with f-strings::
+
+   >>> import datetime
+   >>> now = datetime.datetime.now()
+   >>> f"{now}"
+   '2020-07-19 23:39:01.199675'
+
+Note that this is the :func:`str` format of a :mod:`datetime`. If you want to show the
+:func:`repr` format you use a conversion. There are three conversions beginning with the
+``'!'`` (exclamation) mark:
+
++------------+---------------+
+| Conversion | Meaning       |
++============+===============+
+| ``'!r'``   | :func:`repr`  |
++------------+---------------+
+| ``'!s'``   | :func:`str`   |
++------------+---------------+
+| ``'!a'``   | :func:`ascii` |
++------------+---------------+
+
+In our example of the datetime that becomes::
+
+   >>> f"{now!r}"
+   'datetime.datetime(2020, 7, 19, 23, 39, 1, 199675)'
+   >>> f"{now!a}"
+   'datetime.datetime(2020, 7, 19, 23, 39, 1, 199675)'
```

Comment:
The datetime example is good to show the difference between `!s` and `!r`, but not between `!r` and `!a`, I would use something like:
```python
>>> hello = '你好'
>>> print(f'The ASCII version of {hello!r} is {hello!a}.')
The ASCII version of '你好' is '\u4f60\u597d'.
```

Reply:
Done.